app/router/student.py

```diff
@@ -1,384 +1,309 @@
-            print(enrollment_data)
-            await enrollment_record.create_enrollment_record(
-                build_request(body=enrollment_data)
-            )
 
+        if does_student_already_exist:
             return query_params["student_id"]
 
     else:
-        if (
-            "email" not in query_params
-            or query_params["email"] == ""
-            or query_params["email"] is None
-        ) or (
-            "phone" not in query_params
-            or query_params["phone"] == ""
-            or query_params["phone"] is None
-        ):
-            raise HTTPException(
-                status_code=400, detail="Email/Phone is not part of the request data"
-            )
+        check_if_email_or_phone_is_part_of_request(query_params)
 
-        does_user_already_exist = user.get_users(
+        user_already_exists = user.get_users(
             build_request(
                 query_params={
                     "email": query_params["email"],
                     "phone": query_params["phone"],
                 }
             )
         )
-
-        if not does_user_already_exist:
-            while True:
-                id = id_generation(data)
-                does_student_already_exist = verify_student(
-                    build_request(), student_id=id
-                )
-                if not does_student_already_exist:
-                    response = requests.post(
-                        routes.user_db_url,
-                        params=query_params,
-                        headers=db_request_token(),
-                    )
-                    if response.status_code == 201:
-                        return query_params["student_id"]
-                    raise HTTPException(status_code=500, detail="User not created!")
-
-        else:
+        if user_already_exists:
             response = get_students(
-                build_request(
-                    query_params={"user_id": does_user_already_exist["user_id"]}
-                )
-            )
-            if response.status_code == 200:
-                return response["student_id"]
-            raise HTTPException(
-                status_code=404, detail="User and student details do not match!"
+                build_request(query_params={"user_id": user_already_exists["user_id"]})
             )
+            return response["student_id"]
+
+        else:
+            if data["auth_group"] == "JNVStudents":
+                student_id = generate_JNV_student_id(data)
+
+            student_id_already_exists = verify_student(build_request(), student_id=id)
+
+            if student_id_already_exists:
+                return student_id
+
+    if "grade" in query_params:
+        student_grade_id = grade.get_grade(
+            build_request(query_params={"number": int(query_params["grade"])})
+        )
+        query_params["grade_id"] = student_grade_id["id"]
+
+    if "physically_handicapped" in query_params:
```

Comment:
why this code only for physically handicapped key? can't physically_handicapped be handled before data comes to backend